research/conditional/utils/layer_manager.py

```diff
@@ -60,9 +61,12 @@ def log(self, step):
         ):
             if step % freq == 0:
                 verbosity_levels.append(level)
+        # if we are actually logging, we also want to log the time
+        if len(verbosity_levels) > 0:
+            verbosity_levels.append(-1)
```

Comment:
I don't like this approach. How about we can have different levels of verbosity and higher level (or lower) consist new logs and from higher (or lower) levels